tests/test_planners/rrg_test.py

```diff
@@ -0,0 +1,31 @@
+from gennav.envs import PolygonEnv
+from gennav.planners.rrt.rrg import RRG
+from gennav.utils import RobotState
+from gennav.utils.geometry import Point
+from gennav.utils.samplers import UniformRectSampler
+
+
+def test_rrg_plan():
+    general_obstacles_list = [
+        [[(8, 5), (7, 8), (2, 9), (3, 5)], [(3, 3), (3, 5), (5, 5), (5, 3)]],
+        [
+            [(2, 10), (7, 10), (7, 1), (6, 1), (6, 6), (4, 6), (4, 9), (2, 9)],
+            [(4, 0), (4, 5), (5, 5), (5, 0)],
+            [(8, 2), (8, 7), (10, 7), (10, 2)],
+        ],
+    ]
+
+    sampler = UniformRectSampler(-5, 15, -5, 15)
+    poly = PolygonEnv()
+    for obstacles in general_obstacles_list:
+        poly.update(obstacles)
+
+        start = RobotState(position=Point(0, 0))
+        goal = RobotState(position=Point(12, 10))
+        my_tree = RRG(sampler=sampler, expand_dis=1.0, max_iter=500)
```

Comment:
These lines can be taken out the loop. You can check out the other tests as examples.

Reply:
Line 23 and 24 can be taken out of the loop. If I take the 25th line out, then I am getting the same path for both the obstacles, which leads to an assertion error for the 2nd obstacle. I am still searching why this happens, cuz RRT works fine with this.